eslint.config.js

```diff
@@ -0,0 +1,79 @@
+// eslint.config.js
```

Comment:
What is the advantage of moving to this format, instead of what we had?

Reply:
We had that issue in SkyPortal, latest eslint deprecates the old format, see here: https://eslint.org/blog/2024/04/eslint-v9.0.0-released/#flat-config-is-now-the-default-and-has-some-changes

To be honest I don't like the new format, it's a little confusing and was hard to get it to work, but that's the price to pay to be able to keep eslint up to date going forward.

Reply:
we should probably make that update in baselayer too at some point, but since we have no package.json and very little JS code there, it's probably less relevant

Reply:
It feels to me like they ran in exactly the wrong direction (read the "motivations" here: https://github.com/eslint/rfcs/tree/main/designs/2019-config-simplification#motivation), but oh well.

Reply:
Everything in the Javascript world is WRONG.